core/src/impl/Kokkos_Core.cpp

```diff
@@ -77,6 +78,54 @@ std::stack<hook_function_type, std::list<hook_function_type>> finalize_hooks;
 namespace Kokkos {
 namespace Impl {
 
+ExecSpaceManager ExecSpaceManager::g_space_initializer;
+
+ExecSpaceManager& ExecSpaceManager::get_instance() {
+  return g_space_initializer;
+}
+
+void ExecSpaceManager::register_space_factory(
+    const std::string name, std::unique_ptr<ExecSpaceInitializerBase> space) {
+  printf("registering exec space: %s \n", name.c_str());
```

Comment:
Reason for using `printf` instead of `cout`?

Reply:
resolved with 424fc647